op-batcher/batcher/channel_manager.go

```diff
@@ -97,34 +102,21 @@ func (s *channelManager) TxFailed(_id txID) {
 	if channel, ok := s.txChannels[id]; ok {
 		delete(s.txChannels, id)
 		channel.TxFailed(id)
-		if s.closed && channel.NoneSubmitted() {
-			s.log.Info("Channel has no submitted transactions, clearing for shutdown", "chID", channel.ID())
-			s.removePendingChannel(channel)
-		}
 	} else {
 		s.log.Warn("transaction from unknown channel marked as failed", "id", id)
 	}
 }
 
-// TxConfirmed marks a transaction as confirmed on L1. Unfortunately even if all frames in
-// a channel have been marked as confirmed on L1 the channel may be invalid & need to be
-// resubmitted.
-// This function may reset the pending channel if the pending channel has timed out.
+// TxConfirmed marks a transaction as confirmed on L1. Only if the channel timed out
+// the channelManager's state is modified.
 func (s *channelManager) TxConfirmed(_id txID, inclusionBlock eth.BlockID) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	id := _id.String()
 	if channel, ok := s.txChannels[id]; ok {
 		delete(s.txChannels, id)
-		done, blocksToRequeue := channel.TxConfirmed(id, inclusionBlock)
-		if done {
-			s.removePendingChannel(channel)
-			if len(blocksToRequeue) > 0 {
-				s.blocks.Prepend(blocksToRequeue...)
-			}
-			for _, b := range blocksToRequeue {
-				s.metr.RecordL2BlockInPendingQueue(b)
-			}
```

Comment:
We need to make sure that we continue to track metrics properly, as this is something that has regressed in the past. Ideally, we add some tests to protect this.

Reply:
[4cd021a](https://github.com/ethereum-optimism/optimism/pull/12390/commits/4cd021a06b1712cfb09d4447b199d8d3232f055c)

Reply:
Added coverage for some metrics here [7a45ecd](https://github.com/ethereum-optimism/optimism/pull/12390/commits/7a45ecd795ff99ef3e99b23af0b3caf901b0a20d)